kythe/cxx/indexer/cxx/recursive_type_visitor.h

```diff
@@ -0,0 +1,382 @@
+/*
+ * Copyright 2018 The Kythe Authors. All rights reserved.
```

Comment:
\robinwilliams{What year is it?}

Reply:
lol, in fairness I'm pretty sure that's when I originally wrote this file (I have a bit of a backlog of refactorings).